src/main/kotlin/com/github/julianhille/sqldelight/jscipherdriver/JsSqlCipherDriver.kt

```diff
@@ -0,0 +1,207 @@
+        if (successful) {
+          db.exec("END TRANSACTION")
+        } else {
+          db.exec("ROLLBACK TRANSACTION")
+        }
+      }
+      transaction = enclosingTransaction
+    }
+  }
+
+  private fun createOrGetStatement(identifier: Int?, sql: String): Statement = db.prepare(sql)
+
+  fun getVersion(): Int = (db.pragma("user_version") as? Double)?.toInt() ?: 0
+
+  private fun setVersion(version: Int): Unit {
+    db.pragma("user_version=${version}")
+  }
+
+  fun migrateIfNeeded(
+    create: (SqlDriver) -> Unit,
+    upgrade: (SqlDriver, Int, Int) -> Unit,
+    version: Int
+  ) {
+    val initialVersion = getVersion()
+    println("soimething mumble mumble $initialVersion")
+    println("soimething mumble mumble $db")
+    println("soimething mumble mumble $transaction")
+    val driver = this
+    if (initialVersion == 0) {
+      transaction.run {
+        create(driver)
+        setVersion(version)
+      }
+    } else if (initialVersion != version) {
+      if (initialVersion > version) {
+          throw IllegalStateException("Database version $initialVersion newer than config version $version")
+        }
+        transaction.run {
+          upgrade(driver, initialVersion, version)
+          setVersion(version)
+        }
+    }
+  }
+}
+
+class SqlJsCipherStatement: SqlPreparedStatement {
+  val parameters = mutableListOf<Any?>()
+  override fun bindBytes(index: Int, bytes: ByteArray?) {
+    parameters.add(bytes)
+  }
+
+  override fun bindLong(index: Int, long: Long?) {
+    // We convert Long to Double because Kotlin's Double is mapped to JS number
+    // whereas Kotlin's Long is implemented as a JS object
+    parameters.add(long?.toDouble())
+  }
+
+  override fun bindDouble(index: Int, double: Double?) {
+    parameters.add(double)
+  }
+
+  override fun bindString(index: Int, string: String?) {
+    parameters.add(string)
+  }
+
+  fun bindBoolean(index: Int, boolean: Boolean?) {
+    parameters.add(
+      when (boolean) {
+        null -> null
+        true -> 1.0
+        false -> 0.0
+      }
+    )
+  }
+}
+
+class SqlJsCipherCursor(private val statementIterator: StatementIterator): SqlCursor {
+  var lastResult: IteratorResult? = null
+  var columns: List<String> = listOf<String>()
+
+
+  private fun getIndex(index: Int):  dynamic? /* Number | String | Uint8Array | Nothing? */ {
+    val name = columns[index]
+    val value = lastResult?.value ?: throw NoSuchElementException()
+    return js("value[name]")
+  }
+
+  override fun close() {
+    statementIterator.`return`()
+  }
+
+  override fun getString(index: Int): String? = getIndex(index)
+  override fun getLong(index: Int): Long? = (getIndex(index) as? Double)?.toLong()
+  override fun getBytes(index: Int): ByteArray? = (getIndex(index) as? Uint8Array)?.let {
+    Int8Array(it.buffer).unsafeCast<ByteArray>()
+  }
+
+  override fun getDouble(index: Int): Double? = getIndex(index)
+
+  fun getBoolean(index: Int): Boolean? {
```

Comment:
No override?

https://github.com/cashapp/sqldelight/blob/cb699fcc19632a70deeb2930470bcf09db625a42/drivers/sqljs-driver/src/main/kotlin/app/cash/sqldelight/driver/sqljs/JsSqlDriver.kt#L131-L135

Reply:
does not exist in 1.5.3 i removed it until we need to implement it